snippets/src/reliableTransactionSubmission.ts

```diff
@@ -0,0 +1,200 @@
+ * 
+ * For the implementation in this example, we have made the following decisions:
+ * 1) The script will choose the account sequence and LastLedgerSequence numbers automatically. We allow ripple-lib to choose the fee.
+ *    Payments are defined upfront, and idempotency is not needed. If the script is run a second time, duplicate payments will result.
+ * 2) We will listen for notification that a new validated ledger has been found, and poll for transaction status at that time.
+ *    Futhermore, as a precaution, we will wait until the server is 3 ledgers past the transaction's LastLedgerSequence
+ *    (with the transaction nowhere to be seen) before deciding that it has definitively failed per [3C]
+ * 3) Transactions will not be automatically retried. Transactions are limited to XRP-to-XRP payments and cannot "succeed" in an unexpected way.
+ */
+reliableTransactionSubmissionExample()
+
+async function reliableTransactionSubmissionExample() {
+  /**
+   * Array of payments to execute.
+   * 
+   * For brevity, these are XRP-to-XRP payments, taking a source, destination, and an amount in drops.
+   * 
+   * The script will attempt to make all of these payments as quickly as possible, and report the final status of each. Transactions that fail are NOT retried.
+   */
+  const payments = []
+  
+  const sourceAccount = (await generateTestnetAccount()).account
+  console.log(`Generated new Testnet account: ${sourceAccount.classicAddress}/${sourceAccount.secret}`)
+  // Send amounts from 1 drop to 10 drops
+  for (let i = 1; i <= 10; i++) {
+    payments.push({
+      source: sourceAccount,
+      destination: 'rhsoCozhUxwcyQgzFi1FVRoMVQgk7cZd4L', // Random Testnet destination
+      amount_drops: i.toString(),
+    })
+  }
+  const results = await performPayments(payments)
+  console.log(JSON.stringify(results, null, 2))
+  process.exit(0)
+}
+
+async function performPayments(payments) {
+  const finalResults = []
+  const txFinalizedPromises = []
+  const api = new RippleAPI({server: 'wss://s.altnet.rippletest.net:51233'})
+  await api.connect()
+
+  for (let i = 0; i < payments.length; i++) {
+    const payment = payments[i]
+    const account_info: AccountInfoResponse = await api.request('account_info', {
+      account: payment.source.classicAddress,
+      ledger_index: 'current'})
+    const sequence = account_info.account_data.Sequence
+    const preparedPayment = await api.preparePayment(payment.source.classicAddress, {
+      source: {
+        address: payment.source.classicAddress,
+        amount: {
+          value: payment.amount_drops,
+          currency: 'drops'
+        }
+      },
+      destination: {
+        address: payment.destination,
+        minAmount: {
+          value: payment.amount_drops,
+          currency: 'drops'
+        }
+      }
+    }, {
+      sequence
+    })
+    const signed = api.sign(preparedPayment.txJSON, payment.source.secret)
+    finalResults.push({
+      id: signed.id
```

Comment:
this is really a txHash right?

Reply:
yes

Reply:
Got it. I'd consider renaming it, but no strong opinions since you have more contextual knowledge.